Conch/lib/Conch/Controller/Orc/Lifecycles.pm

```diff
@@ -0,0 +1,134 @@
+=head1 NAME
+
+Conch::Controller::Orc::Lifecycles
+
+=head1 METHODS
+
+=cut
+
+package Conch::Controller::Orc::Lifecycles;
+
+use Mojo::Base 'Mojolicious::Controller', -signatures;
+
+use Conch::Orc;
+
+
+=head2 get_all
+
+Get all lifecycles
+
+=cut
+
+sub get_all ($c) {
+	return $c->status_with_validation(200, 
+		OrcLifecycles => Conch::Orc::Lifecycle->all()
+	);
+}
+
+
+=head2 get_one
+
+Get a single lifecycle
+
+=cut
+
+sub get_one ($c) {
+	my $l = Conch::Orc::Lifecycle->from_id($c->param('id'));
+	return $c->status(404 => { error => "Not found" }) unless $l;
+
+	$c->status_with_validation(200, OrcLifecycle => $l );
+}
+
+
+=head2 create
+
+Create a new lifecycle
+
+=cut
+
+sub create ($c) {
+	return $c->status(403) unless $c->is_global_admin;
+	my $body = $c->validate_input('OrcLifecycleCreate') or return;
+
+	$body->{version} = 0 unless $body->{version};
+	if(Conch::Orc::Lifecycle->from_name_and_version(
+		$body->{name},
+		$body->{version}
+	)) {
+		return $c->status_with_validation(400, Error => {
+			error => "Lifecycle already exists with this name and version"
+		});
+	}
+
+	
+	my $l = Conch::Orc::Lifecycle->new($body->%*)->save();
+
+	$c->status(303 => "/o/lifecycle/".$l->id);
+}
+
+
+=head2 add_workflow
+
+Add a workflow to a lifecycle plan, with an optional plan_order 
+
+=cut
+
+sub add_workflow ($c) {
+	return $c->status(403) unless $c->is_global_admin;
+
+	my $l = Conch::Orc::Lifecycle->from_id($c->param('id'));
+	return $c->status(404 => { error => "Not found" }) unless $l;
+
+	my $body = $c->validate_input('OrcLifecycleAddWorkflow') or return;
+	
+	my $w = Conch::Orc::Workflow->from_id($body->{workflow_id});
+	return $c->status(404 => { error => "Not found" }) unless $w;
```

Comment:
You might distinguish that the workflow ID doesn't exist, to avoid confusion that the URL is incorrect (goes for others in this file).